fuzz/zpay32/encode.go

```diff
@@ -0,0 +1,44 @@
+// +build gofuzz
+
+package zpay32fuzz
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+
+	"github.com/lightningnetwork/lnd/zpay32"
+)
+
+// Fuzz_encode is used by go-fuzz.
+func Fuzz_encode(data []byte) int {
+	inv, err := zpay32.Decode(string(data), &chaincfg.TestNet3Params)
+	if err != nil {
+		return 1
+	}
+
+	// Now encode the invoice out with a static key.
+	testPrivKeyBytes, _ := hex.DecodeString("e126f68f7eafcc8b74f54d269fe206be715000f94dac067d1c04a8ca3b2db734")
+	testPrivKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), testPrivKeyBytes)
+
+	testMessageSigner := zpay32.MessageSigner{
```

Comment:
Maybe we could create the `testMessageSigner` upon package init?

Reply:
Yeah I agree this would be better. I don't think it would help the fuzzer as `go-fuzz` doesn't do everything in-process like libfuzzer and have certain variables tracked across runs.